tsdb/engine/tsm1/engine.go

```diff
@@ -2507,6 +2541,28 @@ func (e *Engine) createCallIterator(ctx context.Context, measurement string, cal
 		return nil, nil
 	}
 
+	// check for optimized series iteration for tsi index
+	if e.index.Type() == tsdb.TSI1IndexName {
+		indexSet := tsdb.IndexSet{Indexes: []tsdb.Index{e.index}, SeriesFile: e.sfile}
+		seriesOpt := opt
+		if len(opt.Dimensions) == 0 && call.Name == "count" {
+			// no point ordering the series if we are just counting all of them
+			seriesOpt.Ordered = false
+		}
+		seriesIterator, err := e.createSeriesIterator(measurement, ref, indexSet, seriesOpt)
+		if err != nil {
+			return nil, err
```

Comment:
The reverse of my other comment above: should we call `Close()` on the `seriesIterator` here? My assumption is not to, because I expect when an error is returned from a creation function, the created object is nil or closed. But since we call `Close()` above in a series cursor, just want to ensure that this is different.

Reply:
`seriesIterator` is always nil on errors